src/ui/public/state_management/state_storage/_tests__/hashed_item_store.js

```diff
@@ -0,0 +1,285 @@
+import expect from 'expect.js';
+import sinon from 'sinon';
+
+import StubBrowserStorage from 'test_utils/stub_browser_storage';
+import HashedItemStore from '../hashed_item_store';
+
+describe('hashedItemStore', () => {
+  describe('#constructor', () => {
+    it('retrieves persisted index from sessionStorage', () => {
+      const sessionStorage = new StubBrowserStorage();
+      sinon.spy(sessionStorage, 'getItem');
+
+      const hashedItemStore = new HashedItemStore(sessionStorage);
+      sinon.assert.calledWith(sessionStorage.getItem, HashedItemStore.PERSISTED_INDEX_KEY);
+      sessionStorage.getItem.restore();
+    });
+
+    it('builds index from persisted index', () => {
+      const sessionStorage = new StubBrowserStorage();
+
+      const persistedItemIndex = {
+        a: {
+          hash: 'a',
+          touched: 1,
+        },
+        b: {
+          hash: 'b',
+          touched: 0,
+        },
+      };
+
+      sessionStorage.setItem(
+        HashedItemStore.PERSISTED_INDEX_KEY,
+        JSON.stringify(persistedItemIndex),
+      );
+
+      const hashedItemStore = new HashedItemStore(sessionStorage);
+
+      expect(hashedItemStore._items).to.eql([persistedItemIndex.b, persistedItemIndex.a]);
+
+      expect(hashedItemStore._itemIndex).to.eql({
+        [persistedItemIndex.a.hash]: persistedItemIndex.a,
+        [persistedItemIndex.b.hash]: persistedItemIndex.b,
+      });
+    });
+  });
+
+  describe('#setItem', () => {
+    describe('if the item exists in sessionStorage', () => {
+      let sessionStorage;
```

Comment:
I much prefer the style used above, where the values are bound and defined in the test rather than shared across tests. Do you prefer one over the other?